src/dscanner/analysis/incorrect_infinite_range.d

```diff
@@ -13,85 +13,73 @@ import dparse.lexer;
 /**
  * Checks for incorrect infinite range definitions
  */
-final class IncorrectInfiniteRangeCheck : BaseAnalyzer
+extern(C++) class IncorrectInfiniteRangeCheck(AST) : BaseAnalyzerDmd!AST
 {
-	alias visit = BaseAnalyzer.visit;
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
 	mixin AnalyzerInfo!"incorrect_infinite_range_check";
 
 	///
-	this(string fileName, bool skipTests = false)
+	extern(D) this(string fileName)
 	{
-		super(fileName, null, skipTests);
+		super(fileName);
 	}
 
-	override void visit(const StructBody structBody)
+	override void visit(AST.StructDeclaration sd)
 	{
 		inStruct++;
-		structBody.accept(this);
+		super.visit(sd);
 		inStruct--;
 	}
 
-	override void visit(const FunctionDeclaration fd)
+	override void visit(AST.ClassDeclaration cd)
 	{
-		if (inStruct > 0 && fd.name.text == "empty")
-		{
-			line = fd.name.line;
-			column = fd.name.column;
-			fd.accept(this);
-		}
+		inStruct++;
```

Comment:
This field name is misleading as you  use it for both structs and classes. Use `inAggregate` or `isAggregateMember` or something that better tells the reader what the field is meant to store.